apollo-router/src/plugins/include_subgraph_errors/tests.rs

```diff
@@ -0,0 +1,507 @@
+use futures::StreamExt;
+use insta::with_settings;
+use serde_json::Map;
+use serde_json::Value;
+use serde_json::json;
+use tower::BoxError;
+
+use super::*; // Import items from mod.rs
+use crate::graphql;
+use crate::plugins::test::PluginTestHarness;
+use crate::services::supergraph; // Required for collect
+
+const PRODUCT_ERROR_RESPONSE: &[&str] = &[
+    r#"{"data":{"topProducts":null},"errors":[{"message":"Could not query products","path":[],"extensions":{"test":"value","code":"FETCH_ERROR", "apollo.subgraph.name": "products"}}]}"#,
+];
+const ACCOUNT_ERROR_RESPONSE: &[&str] = &[
+    r#"{"data":null,"errors":[{"message":"Account service error","path":[],"extensions":{"code":"ACCOUNT_FAIL", "apollo.subgraph.name": "accounts"}}]}"#,
+];
+const VALID_RESPONSE: &[&str] = &[
+    r#"{"data":{"topProducts":[{"upc":"1","name":"Table","reviews":[{"id":"1","product":{"name":"Table"},"author":{"id":"1","name":"Ada Lovelace"}},{"id":"4","product":{"name":"Table"},"author":{"id":"2","name":"Alan Turing"}}]},{"upc":"2","name":"Couch","reviews":[{"id":"2","product":{"name":"Couch"},"author":{"id":"1","name":"Ada Lovelace"}}]}]}}"#,
+];
+const NON_SUBGRAPH_ERROR: &[&str] = &[
+    r#"{"data":{"topProducts":null},"errors":[{"message":"Authentication error","path":[],"extensions":{"test":"value","code":"AUTH_ERROR"}}]}"#,
+];
+const INCREMENTAL_RESPONSE: &[&str] = &[
+    r#"{"data":{"topProducts":null},"errors":[{"message":"Main errors error","path":[],"extensions":{"test":"value","code":"MAIN_ERROR", "apollo.subgraph.name": "products"}}]}"#,
+    r#"{"incremental":[{"data":{"topProducts":null},"errors":[{"message":"Incremental error","path":[],"extensions":{"test":"value","code":"INCREMENTAL_ERROR", "apollo.subgraph.name": "products"}}]}]}"#,
+];
+
+async fn build_harness(
+    plugin_config: &Value,
+) -> Result<PluginTestHarness<IncludeSubgraphErrors>, BoxError> {
+    let mut config = Map::new();
+    config.insert("include_subgraph_errors".to_string(), plugin_config.clone());
+    let config = serde_yaml::to_string(&config).expect("config to yaml");
+    PluginTestHarness::builder().config(&config).build().await
+}
+
+async fn run_test_case(
+    config: &Value,
+    mock_responses: &[&str], // Raw bytes representing the *original* supergraph response
```

Comment:
Technically, if these are "raw bytes" then shouldn't the type be `&[u8]`?

Reply:
Fixed in [b4456d8](https://github.com/apollographql/router/pull/7164/commits/b4456d80e3053413a6ef10d3a58bac87c3b86f72)